src/api/middleware/validators/offer.js

```diff
@@ -130,6 +130,160 @@ const create = useExpressValidators([
         .isArray(),
 ]);
 
+const edit = useExpressValidators([
+    body("_id", ValidationReasons.DEFAULT)
+        .exists().withMessage(ValidationReasons.REQUIRED).bail()
+        .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
+        .custom(async (_id, { req }) => {
+            try {
+                const offer = await (new OfferService()).getOfferById(_id, req.user);
+                if (!offer) throw new Error(ValidationReasons.OFFER_NOT_FOUND(_id));
+                if (req.user?.company.toString() !== offer.owner.toString()) throw new Error(ValidationReasons.NOT_OFFER_OWNER(_id));
+            } catch (_e) {
+                // Also catches any fail to the DB
+                throw new Error(_e);
+            }
+
+            return true;
+        }),
+
+
+    body("title", ValidationReasons.DEFAULT)
+        .optional()
+        .isString().withMessage(ValidationReasons.STRING)
+        .isLength({ max: OfferConstants.title.max_length }).withMessage(ValidationReasons.TOO_LONG(90))
+        .trim(),
+
+    body("publishDate", ValidationReasons.DEFAULT)
+        .optional()
+        .isISO8601({ strict: true }).withMessage(ValidationReasons.DATE).bail()
+        .custom(async (publishDateRaw, { req }) => {
+            try {
+                const offer = await (new OfferService()).getOfferById(req.body._id, req.user);
+
+                const { publishEndDate: publishEndDateCandidate } = req.body;
+
+                const publishDate = (new Date(Date.parse(publishDateRaw))).toISOString();
+
+                if (publishDate <= (new Date(Date.now())).toISOString()) {
+                    throw new Error(ValidationReasons.MUST_BE_AFTER("currentDate"));
+                }
+
+                // If the new publishEndDate is after the new publishDate, the verification will be done in publishEndDate
+                if (publishDate >= offer.publishEndDate.toISOString() &&
+                        !publishEndDateCandidate) {
+
+                    // end date is earlier than publish date, error!
+                    throw new Error(ValidationReasons.MUST_BE_BEFORE("publishEndDate"));
+                }
+
+            } catch (_e) {
+                // Also catches any fail to the DB
+                throw new Error(_e);
+            }
+            return true;
+        })
```

Comment:
This validator got too big and is confusing to read and review (same as publishEndDate one). Consider moving the inner function outside, and even splitting the function in smaller more clear bits